### PR TITLE
URI-encoding for device id in HTTP requests

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: src/iotEdgeExplorer.ts [hunk 1]
```diff
@@ -205,7 +205,7 @@ export class IoTEdgeExplorer extends BaseExplorer {
         this._outputChannel.show();
         this.outputLine(label, `Start deployment to [${deviceId}]`);
 
-        const url = `/devices/${deviceId}/applyConfigurationContent?api-version=${Constants.IoTHubApiVersion}`;
+        const url = `/devices/${encodeURIComponent(deviceId)}/applyConfigurationContent?api-version=${Constants.IoTHubApiVersion}`;
```
Comment (adashen) [inline src/iotEdgeExplorer.ts]: do we need to encodeUri for the whole uri string?

Comment (formulahendry) [reply src/iotEdgeExplorer.ts]: No needed, only `deviceId` would have special char like `#` which needs URI encode

Comment (adashen) [reply src/iotEdgeExplorer.ts]: will # be escaped?

Comment (formulahendry) [reply src/iotEdgeExplorer.ts]: # will be encoded.

**Pull request merged**